### PR TITLE
Add exception middleware

### PR DESCRIPTION
**Pull request by meaboutsoftware**
This PR contains changes related to following areas:

- Exception middleware that handles Business rule validation exceptions and all other exceptions (as internal server error)
- Negative integration tests that were missing for contract preparation
- Negative integration tests that were missing for contract signing

### CODE REVIEW

File: Src/Fitnet/Program.cs [hunk 1]
```diff
@@ -33,6 +34,8 @@
 
 app.UseAuthorization();
 
+app.UseMiddleware<ExceptionMiddleware>();
```
Comment (kamilbaczek) [inline Src/Fitnet/Program.cs:37]: Could you wrap it in extensions method. 

in result we could have.

app.UseErrorHandling();

File: Src/Fitnet/ExceptionMiddleware.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                break;
+            default:
+                statusCode = (int)HttpStatusCode.InternalServerError;
+                message = exception.Message;
+                break;
+        }
+
+        context.Response.StatusCode = statusCode;
+
+        var result = JsonConvert.SerializeObject(new
```
Comment (kamilbaczek) [inline Src/Fitnet/ExceptionMiddleware.cs:47]: Have we benefit from using dynamic over strongly typed record like "internal record ErrorDetails(string Message)"

File: Src/Fitnet.UnitTests/ExceptionMiddlewareTests.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
+namespace SuperSimpleArchitecture.Fitnet.UnitTests;
+
+using System.Net;
+using Microsoft.AspNetCore.Http;
+using Newtonsoft.Json;
+using SuperSimpleArchitecture.Fitnet.Shared.BusinessRulesEngine;
+
+public class ExceptionMiddlewareTests
+{
+    private readonly HttpContext _context;
```
Comment (kamilbaczek) [inline Src/Fitnet.UnitTests/ExceptionMiddlewareTests.cs:10]: could we add enter here?

File: Src/Fitnet.UnitTests/ExceptionMiddlewareTests.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    public ExceptionMiddlewareTests()
+    {
```
Comment (kamilbaczek) [inline Src/Fitnet.UnitTests/ExceptionMiddlewareTests.cs:12]: public ExceptionMiddlewareTests()  => _context = GetHttpContext();

**Review by kamilbaczek — CHANGES_REQUESTED** [review]

File: Src/Fitnet/ExceptionMiddleware.cs (hunk 2 above)
Comment (meaboutsoftware) [reply Src/Fitnet/ExceptionMiddleware.cs:47]: Done

File: Src/Fitnet.UnitTests/ExceptionMiddlewareTests.cs (hunk 4 above)
Comment (meaboutsoftware) [reply Src/Fitnet.UnitTests/ExceptionMiddlewareTests.cs:12]: Done

File: Src/Fitnet.UnitTests/ExceptionMiddlewareTests.cs (hunk 3 above)
Comment (meaboutsoftware) [reply Src/Fitnet.UnitTests/ExceptionMiddlewareTests.cs:10]: Done

File: Src/Fitnet/ExceptionMiddleware.cs [hunk 5]
```diff
@@ -0,0 +1,55 @@
+namespace SuperSimpleArchitecture.Fitnet;
+
+using System.Net;
+using Newtonsoft.Json;
+using Shared.BusinessRulesEngine;
+
+internal class ExceptionMiddleware
```
Comment (kamilbaczek) [inline Src/Fitnet/ExceptionMiddleware.cs:7]: Could be sealed?

**Review by kamilbaczek — CHANGES_REQUESTED** [review]

File: Src/Fitnet/Program.cs (hunk 1 above)
Comment (meaboutsoftware) [reply Src/Fitnet/Program.cs]: Done

**Review by kamilbaczek — APPROVED** [review]

**Pull request merged**